### PR TITLE
fix params + adds missing id from getContact test

### PR DESCRIPTION
**Pull request by averissimo**
tests were not running correctly out of the box

`extend` was overwriting the params set by tests/usage

### CODE REVIEW

**Comment by olalonde** [comment]
Can you give an example of what this fixes? I'm not sure I understand. Also, remove the `console.log` line :)

**Comment by averissimo** [comment]
pulling the master branch and running `npm run test` has this output

all comments in respective lines (see the `Files changed` tab)

image

File: index.js [hunk 1]
```diff
@@ -96,8 +96,7 @@ GoogleContacts.prototype._get = function (params, cb) {
 
 GoogleContacts.prototype.getContacts = function (cb, params) {
     var self = this;
-
-    this._get(_.extend({type: 'contacts'}, params, this.params), receivedContacts);
+    this._get(_.extend({type: 'contacts'}, this.params, params), receivedContacts);
```
Comment (averissimo) [inline index.js:100]: line 100 of `index.js` would remove the id parameter and all others that would be defined in the function call, so a reorder was necessary.

File: index.js [hunk 2]
```diff
@@ -135,7 +134,7 @@ GoogleContacts.prototype.getContact = function (cb, params) {
     function receivedContact(err, contact) {
         if (err) return cb(err);
 
-        cb(null, contact);
+        cb(null, _.get(contact, 'entry', []));
```
Comment (averissimo) [inline index.js:138]: to get the json object specific for a contact, still not as clean as when obtaining all contacts

File: index.js [hunk 3]
```diff
@@ -162,13 +161,15 @@ GoogleContacts.prototype._saveContactsFromFeed = function (feed) {
 GoogleContacts.prototype._buildPath = function (params) {
     if (params.path) return params.path;
 
-    params = _.extend({}, params, this.params);
+    params = _.extend({}, this.params, params);
```
Comment (averissimo) [inline index.js:165]: reorder to allow setting the given params, instead of overwriting them

File: index.js [hunk 4]
```diff
@@ -162,13 +161,15 @@ GoogleContacts.prototype._saveContactsFromFeed = function (feed) {
...
     params.type = params.type || 'contacts';
     params.alt = params.alt || 'json';
     params.projection = params.projection || (params.thin ? 'thin' : 'full');
     params.email = params.email || 'default';
     params['max-results'] = params['max-results'] || 10000;
 
+    console.log('params', params)
+
```
Comment (averissimo) [inline index.js]: oops :)

File: tests/test.js [hunk 5]
```diff
@@ -13,17 +13,26 @@ var c = new GoogleContacts({
 c.getContacts(function (err, contacts) {
   if (err) throw err;
   assert.ok(typeof contacts === 'object', 'Contacts is not an object');
-  console.log(contacts);
+  console.log('Showing first 10 contacts out of '+ contacts.length)
+  console.log(contacts.slice(0,9));
```
Comment (averissimo) [inline tests/test.js:17]: sorry, forgot to remove this... I wanted a cleaner version of the results

before merging I'll return this to original log

File: tests/test.js [hunk 6]
```diff
@@ -13,17 +13,26 @@ var c = new GoogleContacts({
...
+  // setting first contact as the id for the getContact test
   contactsTested = true;
-});
+  test_id = contacts[0].id
```
Comment (averissimo) [inline tests/test.js:20]: `getContact` needs a valid `id`, only way to get it that is not specific for a single account is to retrieve one from the results, that's why the tests are nested

File: tests/test.js [hunk 7]
```diff
@@ -13,17 +13,26 @@ var c = new GoogleContacts({
...
+  c.getContact(function (err, contact) {
+    if (err) throw err;
+    assert.ok(typeof contact === 'object', 'Contact is not an object');
+    console.log('Showing individual contact (first from previous test)')
+    console.log(contact);
+    contactTested = true;
+  }, {
+    id: test_id
+  });
 });
```
Comment (averissimo) [inline tests/test.js:32]: Result of `c.getContact()`

image

File: index.js [hunk 8]
```diff
@@ -97,7 +97,7 @@ GoogleContacts.prototype._get = function (params, cb) {
 GoogleContacts.prototype.getContacts = function (cb, params) {
     var self = this;
 
-    this._get(_.extend({type: 'contacts'}, params, this.params), receivedContacts);
+    this._get(_.extend({type: 'contacts'}, this.params, params), receivedContacts);
```
Comment (averissimo) [inline index.js:100]: this would overide the `params` argument by the default ones

File: tests/test.js (hunk 5 above)
Comment (olalonde) [reply tests/test.js:17]: Oh, I didn't notice that was only in the test file

**Comment by olalonde** [comment]
Thanks. Would you like to maintain this module? I can add you as a contributor on github and npm. I don't really use this module anymore and don't have time to maintain it.

**Comment by averissimo** [comment]
after one pull request? :o 

Sorry man, today was the first time I saw this and probably will be the last.

I was just checking if I could get contacts from hangouts `chat_id`, but unfortunately it can't

Seems like some folks at https://github.com/callinize/google-contacts/ have forked this module and are kinda active (createContact / updateContact / ...)

**Pull request closed without merging**